internal/repositories/sql/dashboards.go

```diff
@@ -0,0 +1,220 @@
+package sql
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+
+	"platform-go-challenge/internal/app/assets"
+	"platform-go-challenge/internal/app/dashboards"
+	"platform-go-challenge/internal/pagination"
+
+	sq "github.com/Masterminds/squirrel"
+)
+
+// Dashboards sql repository struct.
+type Dashboards struct {
+	dbClient BasicConnectionWithTransactions
+}
+
+// NewDashboardsRepository constructor.
+func NewDashboardsRepository(dbClient BasicConnectionWithTransactions) *Dashboards {
+	return &Dashboards{dbClient: dbClient}
+}
+
+// GetUserDashboardID returns id of a user's dashboard.
+func (d *Dashboards) GetUserDashboardID(ctx context.Context, userID uint32) (uint32, error) {
+	dashboardID := uint32(0)
+	err := sq.Select("d.id").
+		From(dashboardsTable + " AS d").
+		Where(sq.Eq{"d.user_id": userID}).
```

Comment:
Could that lead to SQL Injection?

Reply:
I purposely used this package since it's easier to build a safe sql query. It keeps all arguments in a slice and uses placeholders in the generated query.